src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementFhirImportService.cs

```diff
@@ -18,13 +17,12 @@
 using Microsoft.Health.Fhir.Ingest.Telemetry;
 using Microsoft.Health.Fhir.Ingest.Template;
 using Microsoft.Health.Logging.Telemetry;
-using Microsoft.Toolkit.HighPerformance;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Health.Fhir.Ingest.Service
 {
-    public class MeasurementFhirImportService : ParallelTaskWorker<MeasurementFhirImportOptions>
+    public class MeasurementFhirImportService : ParallelTaskWorker<MeasurementFhirImportOptions>, IImportService

```

Comment:
One minor change to guard against potential data loss if the Normalization is switched to compression but the FHIR Transformation isn't.

In the Linq query of ParseEventData where we are parsing events, if the content type is set then throw an exception since this class doesn't know how to handle it.  I would say throw a not supported exception but I suspect we handle and ignore those so we might need to use a new type.

Reply:
maybe we should throw a `DefinitelyNotSupportedException`?

kidding. added CompressionNotSupportedException to MeasurementFhirImportService.cs and added a new unit test for this scenario.